docs/basics/101-108-run.rst

```diff
@@ -0,0 +1,196 @@
+from the shell.
+
+Lets say, for example, that you enjoyed the longnow podcasts a lot,
+and you start a podcast-night with friends to wind down from all of
+the exciting DataLad lectures. They propose to make a
+list of speakers and titles to cross out what they've already listened
+to, and ask you to prepare such a list.
+
+"Mhh... probably there is a datalad way to do this... wasn't there also
+a note about meta data extraction at some point?" But as we're not that
+far into the lectures, you decide to write a simple Python script
+to generate a tab-separated file that lists speaker and title
+name instead.
+
+To do this, we're following a best practice that will reappear in the
+later section on YODA principles (todo: link): Collecting all
+additional scripts that work with content of a subdataset *outside*
+of this subdataset, in a dedicated ``code/`` directory,
+and collating the output of the execution of these scripts
+*outside* of the subdataset as well -- and
+therefore not modifying the subdataset.
+
+The motivation behind this will become clear in later sections,
+but for now we'll start with best-practice building.
+Therefore, create a subdirectory ``code/`` in the ``DataLad-101``
+superdataset:
+
+.. runrecord:: _examples/DL-101-108-101
+   :language: console
+   :workdir: dl-101/DataLad-101
+
+   $ mkdir code
+   $ tree -d
+
+Inside of ``Datalad-101/code``, create a simple Python script ``list_titles.py``.
+This script will carry out a simple task:
+It will loop through the file names of the ``.mp3`` files and
+write out speaker names and talk titles in a very simple fashion.
+The content of this script is written below -- the ``cat`` command
+will write it into the script.
+
+.. runrecord:: _examples/DL-101-108-102
+   :language: console
+   :workdir: dl-101/DataLad-101
+
+   $ cat << EOT > code/list_titles.py
+   # import modules for path operations and pathname pattern expansion
+   from os.path import join, basename
+   from glob import glob
+
+   # define a path to the podcasts. Note that this is a relative path,
+   # pointing from the root of DataLad-101 to the podcasts:
+
+   path = join('recordings', 'longnow',
+               'Long_Now__Seminars_About_Long_term_Thinking', '*')
+   # create a list of all file names found under the path
+   files = glob(path)
+   # define an outfile name:
+   outfile = join('recordings', 'Podcasts.tsv')
+
+   # loop through all file names and write speaker and title into the file
+   with open(outfile, 'w') as f:
+       for file in files:
+           # extract speaker name
+           author = basename(file).split('__')[1] \
+               if len(basename(file).split('__')[1].split('_')) > 1 \
+               else '__'.join(basename(file).split('__')[1:3])
+           # extract title
+           title = '_'.join(basename(file).split(author)[-1].split('__')[1:])
+           # write speaker and talk title into the file
+           f.write(author + '\t' + title + '\n')
+   EOT
```

Comment:
Thanks for bringing this up. My shell scripting abilities are quite limited still. Maybe something like this is sufficient?
```
for i in recordings/longnow/Long_Now__Seminars_About_Long_term_Thinking/*__*; do echo $(basename "$i") | awk -F"__" '{print$1 "\t" $2 "\t" $3}' >> recordings/podcasts.tsv; done 
```

Will result in

```
2003_11_15      Brian_Eno       The_Long_Now.mp3
2003_12_13      Peter_Schwartz  The_Art_Of_The_Really_Long_View.mp3
2004_01_10      George_Dyson    There_s_Plenty_of_Room_at_the_Top
2004_02_14      James_Dewar     Long_term_Policy_Analysis.mp3
2004_03_13      Rusty_Schweickart       The_Asteroid_Threat_Over_the_Next_100_000_Years.mp3
2004_04_10      Daniel_Janzen   Third_World_Conservation
2004_05_15      David_Rumsey    Mapping_Time.mp3
2004_06_12      Bruce_Sterling  The_Singularity
2004_07_10      Jill_Tarter     The_Search_for_Extra_terrestrial_Intelligence
2004_08_14      Phillip_Longman The_Depopulation_Problem.mp3
2004_09_11      Danny_Hillis    Progress_on_the_10_000_year_Clock.mp3
```

Any shell scripting advice is welcome ;-)

Reply:
On second thought, that was stupid -- a shell command wouldn't be able to show case a ``datalad rerun``, correct? If we implement a "bug" into the shell command, a "fix" would require a new run instead of a ``rerun``...

Reply:
Yes, a plain command isn't a good replacement. But a shell script file would do it.